File: src/LCR136.shan-chu-lian-biao-de-jie-dian-lcof.100299/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode deleteNode(ListNode head, int val) {
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode prev = dummy;
        while (head != null){
            if(head.val == val){
                prev.next = head.next;
                prev = prev.next;
                head = head.next;
            }else{
                prev = head;
                head = head.next;
            }
        }
        return dummy.next;
    }
}

File: src/LCR136.shan-chu-lian-biao-de-jie-dian-lcof.100299/solution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var deleteNode = function (head, val) {
    const dummyHead = new ListNode();
    dummyHead.next = head;
    let prev = dummyHead;
    while (head && head.val !== val) {
        prev = prev.next;
        head = head.next;
    }
    prev.next = head.next;
    return dummyHead.next;
};
